---
title: "RAB234_summary_report1"
author: "Marion Hardy"
date: "2023-05-31"
output: 
  html_document:
    toc: true 
    theme: spacelab 
    highlight: monochrome
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, cache = TRUE, echo = FALSE, warning = F, cache.lazy = F)
knitr::opts_chunk$set(fig.width=10, fig.height=15) 

library(cowplot)
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
library(tidyverse)
library(ggrepel)
library(clusterProfiler)
library(msigdbr)
library(readxl)
library(eulerr)
library(NMF)
library(pheatmap)
library(openxlsx)
library(fgsea)

```


```{r Data loading, include=FALSE}

#all
counts <- as.data.frame(read.csv("./data/RAB234_counts.csv", row.names = 1))
coldata = read.csv("./data/coldata.csv")

# trpn vs trpp
#nonuc
res_tbl <- read_csv("./data_output/Nonuc_Trpn_vs_Trpp/res_tbl.csv")
diff <- read_csv("./data_output/Nonuc_Trpn_vs_Trpp/res_tbl_signif.csv")
res_tbl1 <-  res_tbl %>% filter(!is.na(ENTREZID), padj<0.05)

nonuc_trpn_trpp <- read_csv("./data_output/Nonuc_Trpn_vs_Trpp/res_tbl.csv")
nonuc_trpn_trpp1 <-  nonuc_trpn_trpp %>% filter(!is.na(ENTREZID), padj<0.05)

#mrpl
mrpl_trpn_trpp =  read_csv("./data_output/Mrpl_Trpn_vs_Trpp/res_tbl.csv")
mrpl_trpn_trpp1 = mrpl_trpn_trpp %>% filter(!is.na(ENTREZID), padj<0.05)

#nduf
nduf_trpn_trpp =  read_csv("./data_output/Nduf_Trpn_vs_Trpp/res_tbl.csv")
nduf_trpn_trpp1 = nduf_trpn_trpp %>% filter(!is.na(ENTREZID), padj<0.05)

# trpp
#mrpl
trpp_mrpl <- read_csv("./data_output/Trpp_Mrpl_vs_Scr/res_tbl.csv")
trpp_mrpl1 <-  trpp_mrpl  %>% filter(!is.na(ENTREZID), padj<0.05)

#nduf
trpp_nduf <- read_csv("./data_output/Trpp_Nduf_vs_Scr/res_tbl.csv")
trpp_nduf1 <-  trpp_nduf %>% filter(!is.na(ENTREZID), padj<0.05)

#nonuc
trpp_nonuc <- read_csv("./data_output/Trpp_Nonuc_vs_Scr/res_tbl.csv")
trpp_nonuc1 <-  trpp_nduf %>% filter(!is.na(ENTREZID), padj<0.05)

# trpn
#mrpl
trpn_mrpl <- read_csv("./data_output/Trpn_Mrpl_vs_Scr/res_tbl.csv")
trpn_mrpl1 <-  trpn_mrpl %>% filter(!is.na(ENTREZID), padj<0.05)

#nduf
trpn_nduf <- read_csv("./data_output/Trpn_Nduf_vs_Scr/res_tbl.csv")
trpn_nduf1 <-  trpn_nduf %>% filter(!is.na(ENTREZID), padj<0.05)

#nonuc
trpn_nonuc <- read_csv("./data_output/Trpn_Nonuc_vs_Scr/res_tbl.csv")
trpn_nonuc1 <-  trpp_nduf %>% filter(!is.na(ENTREZID), padj<0.05)

```

# Introduction

Data from  RAB : CD8+ T cells from mice

- Cell types: not nucleofected, scramble, Nduf KO, Mrpl28 KO

-   Conditions: treated w/ or w/o Tryptophan (70 uM vs 4 uM)

-   Biological replicates (3)

Report based off of a conversation to tailor the analysis more to Raphaëlle's needs.

# Sample similarity

```{r, fig.height=6, fig.width=10}

counts = counts[,-1]

dds <- DESeqDataSetFromMatrix(countData = counts, colData = coldata,
                              design = ~condition+celltype) 

# Generate a linear model

dds$condition <- relevel(dds$condition, "Trpp")
dds <- DESeq(dds)

# Checking PCA

rld <- rlogTransformation(dds)

p1 <- plotPCA(rld,intgroup="condition") # sample switch up?
p2 <- plotPCA(rld,intgroup="celltype") 
p3 <- plotPCA(rld,intgroup=c("condition","celltype")) # determine which samples to relabel

plot_grid(p1, p2, p3, nrow = 2, align = "v")

```

```{r, fig.height=6, fig.width=8, eval=FALSE}

# Checking sample similarity

sampleDists <- dist(t(assay(dds)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(dds$condition, dds$celltype, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

# All violin plots

```{r, fig.width=12, fig.height=8}

p1 =
  res_tbl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp- vs Trp+ in Not nucleofected cells")+
  theme_bw()

p2 =
  trpn_mrpl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp- Mrpl28 vs Scramble")+
  theme_bw()

p3 =
  trpp_mrpl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp+ Mrpl28 vs Scramble")+
  theme_bw()

p4 =
  trpn_nduf %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp- Ndufa2 vs Scramble")+
  theme_bw()

p5 =
  trpp_nduf %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp+ Ndufa2 vs Scramble")+
  theme_bw()

p6 =
  trpn_nonuc %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp- Nonuc vs Scramble")+
  theme_bw()

p7 =
  trpp_nonuc %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Trp+ Nonuc vs Scramble")+
  theme_bw()

p8 =
  mrpl_trpn_trpp %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Mrpl28 KO Trp- vs Trp+")+
  theme_bw()

p9 =
  nduf_trpn_trpp %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey50',
                  fontface = "italic")+
  labs(title = "Ndufa2 KO Trp- vs Trp+")+
  theme_bw()

p1

```

```{r, fig.width=12, fig.height=12}

plot_grid(p2, p3, p4, p5, p6, p7, p8, p9, nrow = 4, align = "v")

```

# Nonuc Trpn vs Trpp
## Volcano plot 

```{r, fig.height = 15, fig.width = 15}

res_tbl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey80',
                  fontface = "italic")+
  labs(title = "Trp- vs Trp+ in Not nucleofected cells")+
  theme_bw()+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), 
       "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_volcano.svg", 
       dpi = 300,
       height = 10, width = 15, device = "svg")

```

Getting some stats yay

```{r, echo=TRUE}

table(res_tbl$padj<=0.05) # significant genes

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange<=-1) # signif and downreg

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange>=1) # signif and upreg

```

## GSEA analysis logFc

```{r}

ordered_genes_fc <- res_tbl1$log2FoldChange
names(ordered_genes_fc) <- res_tbl1$gene
ordered_genes_fc <- sort(ordered_genes_fc, decreasing = T)

```

```{r}

# Load the enrichment data sets

mm_GO_sets <- msigdbr(
  species = "Mus musculus", 
  category = "C5",
  subcategory = "BP") # for gene ontology collection

mm_reactome_sets <- msigdbr(
  species = "Mus musculus", 
  category = "C2",
  subcategory = "CP:REACTOME") # for reactome collection

mm_wiki_sets <- msigdbr(
  species = "Mus musculus", 
  category = "C2",
  subcategory = "CP:WIKIPATHWAYS") # for Wikipathways collection

mm_pert_sets <- msigdbr(
  species = "Mus musculus", 
  category = "C2",
  subcategory = "CGP")

mm_hallmark_sets <- msigdbr(
  species = "Mus musculus", 
  category = "H") # for hallmark collection

# Get only the ones with "glycoly" in them
mm_GO_gl = 
  mm_GO_sets %>% 
  filter(grepl("glycoly", gs_name, ignore.case = T)) %>% 
  rbind(mm_hallmark_sets %>% filter(grepl("glycoly", gs_name, ignore.case = T))) %>% 
  rbind(mm_reactome_sets %>% filter(grepl("glycoly", gs_name, ignore.case = T)))

# Get the ones with respiration and respiration in them
mm_GO_ox  = 
  mm_GO_sets %>% 
  filter(grepl("oxidative", gs_name, ignore.case = T)|
         grepl("respiration", gs_name, ignore.case = T)) %>% 
  rbind(mm_hallmark_sets %>% filter(grepl("oxidative", gs_name, ignore.case = T),
         grepl("respiration", gs_name, ignore.case = T)) %>% 
  rbind(mm_reactome_sets %>% filter(grepl("oxidative", gs_name, ignore.case = T))))

# Get the ones with T cells in them
mm_GO_tcell = 
  mm_GO_sets %>% 
  filter(grepl("t_cell", gs_name, ignore.case = T)) 

# Make a master set for GSEA with the ones we are interested in 
mm_summary = 
  mm_GO_sets %>% 
  filter(gs_name %in% toupper(c("gobp_ribosome_biogenesis",
                        "gobp_ribonucleoprotein_complex biogenesis", 
                        "gobp_ncrna_metabolic_process","gobp_mitochondrial_gene_expression",
                        "gobp_mitochondrial_translation", "gobp_trna metabolic_process",
                        "gobp_trna metabolic_process",
                        "gobp_inflammatory_response","gobp_amino_acid_activation"))) %>% 
  rbind(mm_reactome_sets %>% filter(
    gs_name%in% toupper(c("REACTOME_RESPONSE_OF_EIF2AK1_HRI_TO_HEME_DEFICIENCY",
                          "reactome_translation",
                          "reactome_mitochondrial_translation",
                          "reactome_cholesterol_biosynthesis",
                          "reactome_oxidative_stress_induced_senescence",
                          "reactome_perk_regulates_gene_expression",
                          "reactome_regulation_of_cholesterol_biosynthesis_ny_srebp_srebf",
                          "reactome_dna_methylation",
                          "REACTOME_ATF4_ACTIVATES_GENES_IN_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS",
                          "reactome_sirt1_negatively_regulates_rrna_expression")))) %>% 
  rbind(mm_hallmark_sets %>%  filter(gs_name %in% toupper(c("hallmark_unfolded_protein_response",
                                                            "hallmark_myc_targets_v1",
                                                            "hallmark_cholesterol_homeostasis"))))

# Make a master set for ORA with the ones we are interested in 

mm_ora_summary =
  mm_GO_sets %>% 
  filter(gs_name %in% toupper(c("gobp_t_cell_activation",
                                "gobp_antigen_processing_and_presentation",
                                "gobp_carbohydrate_catabolic_process",
                                "gobp_positive_regulation_of_glycolytic_process",
                                "gobp_regulation_of_antigen_processing_and_presentation",
                                "gobp_regulation_of_atp_metabolic_process",
                                "gobp_regulation_of_t_cell_activation",
                                "gobp_calcium_mediated_signaling",
                                "gobp_pyruvate_metabolic_process",
                                "gobp_cellular_response_to_calcium_ion",
                                "gobp_negative_regulation_of_leukocyte_apoptotic_process",
                                "gobp_unsaturated_fatty_acid_biosynthesis_process",
                                "gobp_regulation_of_glycolytic_process",
                                "gobp_t_cell_differentiation",
                                "gobp_adp_metabolic_process",
                                "GOBP_STEROL_METABOLIC_PROCESS")))
# Make one with all gene sets

mm_all = 
  mm_GO_sets %>% 
  rbind(mm_reactome_sets) %>% 
  rbind(mm_hallmark_sets) %>% 
  rbind(mm_wiki_sets)
 
```

### Summary

```{r, fig.height=6, fig.width=7}

gsea_results <- GSEA(
  geneList = ordered_genes_fc, 
  pvalueCutoff = 1, 
  eps = 0, 
  pAdjustMethod = "BH", 
  nPermSimple = 10000,
  TERM2GENE = dplyr::select(
    mm_summary,
    gs_name,
    gene_symbol
  )
)

dotplot(gsea_results, x = "NES", showCategory = 30, font.size = 7)+ 
  labs(title = "GSEA",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), 
       "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_gsea.svg", 
       dpi = 300,
       height = 6, width = 5, device = "svg")

gsea_results%>% 
  as_tibble() %>% 
  select(ID, setSize,	enrichmentScore,	NES,	pvalue,	p.adjust,	qvalue) %>%
  knitr::kable()

write.xlsx(gsea_results, "./data_output/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_gsea.xlsx")

```


```{r, eval=FALSE}
cnetplot(gsea_results, foldChange=ordered_genes_fc, showCategory = 50,
         cex_label_category = .5, cex_label_gene = .5)
```

### Get the enrichment plots

```{r, fig.height = 2, fig.width = 6}

# For loop to get the GSEA curves and add an annotation with the p-value
# Save them in a figures subfolder

for (i in 1:length(gsea_results@result)){
  term = gsea_results@result$ID[i]
  p = plotEnrichment(gsea_results[[term]],
               ordered_genes_fc) + 
    labs(title=paste0(term))+
    theme(plot.title=element_text(size=8,face="bold"),
          axis.title=element_text(size=10))+
  annotate(geom="text", x=6000, y=0.5, 
           label=paste0("p-value = ",round(gsea_results[term]$p.adjust,5)),size = 2.5)
  print(p)
  ggsave(plot = p, paste0("./figures/Nonuc_Trpn_vs_Trpp/",noquote(term),".svg"),
         device = "svg",dpi = 300, height = 3, width = 4)
}

```

### Get the enrichment plots for glycolysis and oxphos

```{r, fig.height = 2, fig.width = 6}

gsea_results2 <- GSEA(
  geneList = ordered_genes_fc, 
  pvalueCutoff = 1, 
  eps = 0, 
  pAdjustMethod = "BH", 
  nPermSimple = 10000,
  TERM2GENE = dplyr::select(
    mm_all,
    gs_name,
    gene_symbol
  )
)

# Do the same thing as the for loop but by hand for the plots where the pval is
# Not significant

plotEnrichment(gsea_results2[["GOBP_OXIDATIVE_PHOSPHORYLATION"]],
               ordered_genes_fc) + 
    labs(title="GOBP_OXIDATIVE_PHOSPHORYLATION")+
    theme(plot.title=element_text(size=8,face="bold"),
          axis.title=element_text(size=10))+
  annotate(geom="text", x=6000, y=0.5, 
           label=paste0("p-value = ",
                        round(gsea_results2["GOBP_OXIDATIVE_PHOSPHORYLATION"]$p.adjust,5)),
           size = 2.5)

ggsave(plot = last_plot(),
       "./figures/Nonuc_Trpn_vs_Trpp/GOBP_OXIDATIVE_PHOSPHORYLATION.svg",
       device = "svg",dpi = 300, height = 3, width = 4)

plotEnrichment(gsea_results2[["HALLMARK_GLYCOLYSIS"]],
               ordered_genes_fc) + 
    labs(title="HALLMARK_GLYCOLYSIS")+
  theme(plot.title=element_text(size=8,face="bold"),
          axis.title=element_text(size=10))+
  annotate(geom="text", x=6000, y=0.5, 
           label=paste0("p-value = ",
                        round(gsea_results2["GOBP_OXIDATIVE_PHOSPHORYLATION"]$p.adjust,2)),
           size = 2.5)

ggsave(plot = last_plot(),
       "./figures/Nonuc_Trpn_vs_Trpp/HALLMARK_GLYCOLYSIS.svg",
       device = "svg",dpi = 300, height = 3, width = 4)

```

Those plots have a non significant adjusted pvalue and a bad qvalue!!

### Summary breakdown per function

```{r, fig.height=5, fig.width=7}

gsea_results_mitoch = gsea_results %>% 
  filter(ID%in% toupper(c("gobp_ribosome_biogenesis",
                                "gobp_mitochondrial_translation",
                                "gobp_mitochondrial_gene_expression",
                                "reactome_translation",
                                "gobp_inflammatory_response",
                                "hallmark_cholesterol_homeostasis",
                                "reactome_cholesterol_biosynthesis",
                                "gobp_ncran_metabolic_process",
                                "reactome_dna_methylation")))


dotplot(gsea_results_mitoch, x = "NES", showCategory = 30, font.size = 7)+ 
  labs(title = "GSEA : biological processes",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_gsea_bp.svg", 
       dpi = 300,
       height = 3.5, width = 5, device = "svg")
```

```{r, fig.height=4, fig.width=7}

gsea_results_meta = gsea_results %>% 
  filter(ID%in% toupper(c("gobp_amino_acid_activation",
                                "reactome_cholesterol_biosynthesis",
                                "hallmark_cholesterol_homeostasis")) )


dotplot(gsea_results_meta, x = "NES", showCategory = 30, font.size = 7)+ 
  labs(title = "GSEA : metabolic processes",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_gsea_meta.svg", 
       dpi = 300,
       height = 3, width = 5, device = "svg")
```

```{r, fig.height=5, fig.width=7}

gsea_results_regul = gsea_results %>% 
  filter(ID%in% toupper(c("reactome_sirt1_negatively_regulates_rrna_expression",
                                "reactome_response_of_eif2ak1_hri_to_heme_deficiency",
                                "hallmark_unfolded_protein_response",
                                "reactome_perk_regulates_gene_expression",
                                "hallmark_myc_targets_v1",
                                "reactome_atf4_activates_genes_in_response_to_endoplasmic_reticulum_stress")))


dotplot(gsea_results_regul, x = "NES", showCategory = 30, font.size = 7)+ 
  labs(title = "GSEA : signalisation pathways",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_gsea_reg.svg", 
       dpi = 300,
       height = 4, width = 5, device = "svg")
```

## ORA
### Summary

```{r, fig.height = 5, fig.width = 7}

up = res_tbl1 %>% filter(abs(log2FoldChange) >= 1)
up = up$gene

ora_results <- enricher(
  gene = up, 
  pvalueCutoff = 0.05, 
  pAdjustMethod = "BH", 
  TERM2GENE = dplyr::select(
    mm_GO_sets,
    gs_name,
    gene_symbol
  )
)

targetora = toupper(c("gobp_t_cell_activation",
                                "gobp_carbohydrate_catabolic_process",
                                "gobp_positive_regulation_of_glycolytic_process",
                                "gobp_regulation_of_atp_metabolic_process",
                                "gobp_calcium_mediated_signaling",
                                "gobp_pyruvate_metabolic_process",
                                "gobp_cellular_response_to_calcium_ion",
                                "gobp_negative_regulation_of_leukocyte_apoptotic_process",
                                "gobp_unsaturated_fatty_acid_biosynthesis_process",
                                "gobp_regulation_of_glycolytic_process",
                                "gobp_t_cell_differentiation",
                                "gobp_adp_metabolic_process",
                                "GOBP_STEROL_METABOLIC_PROCESS"))

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  dotplot(showCategory = 30, font.size = 7)+ 
  labs(title = "Over-representation analysis",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), 
       "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_ora_dotplot.svg", 
       dpi = 300,
       height = 5, width = 5, device = "svg")

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  barplot(showCategory = 30, font.size = 7)+ 
   labs(title = "Over-representation analysis",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), 
       "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_ora_barplot.svg", 
       dpi = 300,
       height = 5, width = 5, device = "svg")
 
ora_results %>%
  filter(ora_results@result$ID %in% targetora) %>% 
  as_tibble() %>%
  select(ID, GeneRatio,	p.adjust,	qvalue) %>%
  knitr::kable()

write.xlsx(ora_results, "./data_output/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_ora.xlsx")

```

### Summary breakdown per function

```{r, fig.height=5, fig.width=7}

ora_meta = ora_results %>% 
  filter(ID%in% toupper(c( "gobp_carbohydrate_catabolic_process",
                                "gobp_positive_regulation_of_glycolytic_process",
                                "gobp_regulation_of_atp_metabolic_process",
                                "gobp_calcium_mediated_signaling",
                                "gobp_pyruvate_metabolic_process",
                                "gobp_cellular_response_to_calcium_ion",
                                "gobp_unsaturated_fatty_acid_biosynthesis_process",
                                "gobp_regulation_of_glycolytic_process",
                                "gobp_adp_metabolic_process",
                                "GOBP_STEROL_METABOLIC_PROCESS")))

dotplot(ora_meta, showCategory = 30, font.size = 7)+ 
  labs(title = "ORA : metabolic processes",
              subtitle = "Trp 4 µM vs Trp 70 µM for 72h")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), 
       "./figures/Nonuc_Trpn_vs_Trpp/Nonuc_Trpp_vs_Trpn_ora_meta.svg", 
       dpi = 300,
       height = 5, width = 5, device = "svg")
```


## qPCR results found in RNAseq?

```{r, fig.width=3, fig.height=2}

# Get the targets that were looked at by qPCR by Raphaële

glyc_targ = res_tbl1 %>% 
  filter(gene%in%c("Hk2","Ldha","Pdha","Pdk1"))

# Volcano plot with highlight

res_tbl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) > 1, 
             label = ifelse(padj<0.05&log2FoldChange>=1|
                              padj<0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "lightcoral")) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  theme_bw()+
  geom_point(data=glyc_targ, 
             aes(x=log2FoldChange,y=-log10(padj)), 
             color='blue',
             size=0.5)+
  geom_text_repel(data = glyc_targ, size = 3, segment.color = "blue",
                  max.overlaps = 30, min.segment.length = 0, color = "blue",
                  box.padding = 0.5,
                  fontface = "italic")+
  theme(legend.title= element_blank())


glyc_targ %>%
  select(ensembl, baseMean, log2FoldChange, padj, gene, description) %>% 
  knitr::kable()

```

## Heatmap : gene expression per condition

```{r}

dds = readRDS("./data_output/Nonuc_Trpp_vs_Trpn/dds.rds")
diff = read.csv("./data_output/Nonuc_Trpp_vs_Trpn/res_tbl_signif.csv")

reslog = rlog(dds, blind = FALSE) # Do the regularized log transformation from the DESeq2                                       object
mat = assay(reslog) # Get the counts

coldata = as.data.frame(colData(reslog))
colnames(mat) = rownames(coldata)

# Do the scaling to get z-scores

basemean = rowMeans(mat)
mats = t(apply(mat, 1, scale))
colnames(mats) = colnames(mat)

mats = mats[diff$ensembl,] # Keep only the significantly deferentially expressed genes
rownames(mats) = diff$gene
head(mats)

```

These plots are showing a z-score.
If you did the mean(trpn ratio)/mean(trpp ratio), you would get the logFc of the volcano plot

Genes from the ORA results :
- GOBP Positive regulation of glycolytic process (10 genes)
- GOBP Regulation of ATP metabolic process (25)

Those two terms have most genes in common.

```{r}
c("Prxl2c", "Entpd5", "Hif1a", "P2rx7", "App", "Mlxipl",
  "Htr2a", "Prkaa2", "Prkaa1","Slc4a4", "Gapdhs","Nupr1", "Prxl2c", 
  "Shmt2", "Entpd5", "Dnajc15", "Hdac4", "Sphk2","Hif1a", "Slc2a6", 
  "Trp53", "Tspo", "Ak4", "Slc25a33", "Slc25a23", "P2rx7","Gck", 
  "App", "Mlxipl", "Cbfa2t3", "Htr2a", "Cox7a1","Prkaa2", "Pid1", 
  "Slc4a4", "Gapdhs", "Atf4")
```


```{r, fig.width=6, fig.height=8}

pos_reg_gly = ora_results %>%
  as_tibble() %>%
  filter(ID == "GOBP_POSITIVE_REGULATION_OF_GLYCOLYTIC_PROCESS") %>%
  select(geneID)

temp = pos_reg_gly$geneID
temp = gsub("/"," ", temp)

unlist(c(strsplit(ora_results_react1$geneID[[1]],"/")))

reg_atp = ora_results %>%
  as_tibble() %>%
  filter(ID == "GOBP_REGULATION_OF_ATP_METABOLIC_PROCESS") %>%
  select(geneID)

temp = reg_atp$geneID
temp = gsub("/"," ", temp)

# select only the targets to show on the heatmap

m = mats[rownames(mats)%in%c("Prxl2c", "Entpd5", "Hif1a", "P2rx7", "App", "Mlxipl",
                             "Htr2a", "Prkaa2", "Prkaa1","Slc4a4", "Gapdhs",
                             "Nupr1", "Prxl2c", "Shmt2", "Entpd5", "Dnajc15", "Hdac4", "Sphk2",
                             "Hif1a", "Slc2a6", "Trp53", "Tspo", "Ak4", "Slc25a33", "Slc25a23",
                             "P2rx7","Gck", "App", "Mlxipl", "Cbfa2t3", "Htr2a", "Cox7a1",
                             "Prkaa2", "Pid1", "Slc4a4", "Gapdhs", "Atf4"),]

targets = res_tbl %>% filter(gene%in%c("Prxl2c", "Entpd5", "Hif1a", "P2rx7", "App", "Mlxipl",
                             "Htr2a", "Prkaa2", "Prkaa1","Slc4a4", "Gapdhs",
                             "Nupr1", "Prxl2c", "Shmt2", "Entpd5", "Dnajc15", "Hdac4", "Sphk2",
                             "Hif1a", "Slc2a6", "Trp53", "Tspo", "Ak4", "Slc25a33", "Slc25a23",
                             "P2rx7","Gck", "App", "Mlxipl", "Cbfa2t3", "Htr2a", "Cox7a1",
                             "Prkaa2", "Pid1", "Slc4a4", "Gapdhs", "Atf4"))

# aheatmap(m, color = rev(brewer.pal(9,"RdBu")), annColors = "Set2", scale = "none")

newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="Glycolysis and ATP metabolism heatmap", 
         cluster_cols=T,fontsize_row=8, border_color=NA,
         labels_row = as.expression(newnames))

```

```{r, fig.height=4, fig.width=6}

res_tbl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) > 1, 
             label = ifelse(padj<0.05&log2FoldChange>=1|
                              padj<0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "lightcoral")) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  theme_bw()+
  geom_point(data=targets, 
             aes(x=log2FoldChange,y=-log10(padj)), 
             color='blue',
             size=0.5)+
  geom_text_repel(data = targets, size = 3, segment.color = "blue",
                  max.overlaps = 30, min.segment.length = 0, color = "blue",
                  box.padding = 0.5,
                  fontface = "italic")+
  theme(legend.title= element_blank())

```

## Heatmap with complete list of glycolysis genes

After a meeting, we wanted to check the expression of a list of more genes more largely
implicated in glycolysis. This was also added to the KO analyses.
At the end of the document, there are the heatmaps with those genes + OXPHOS and logFc between conditions (and not triplicates with z-scores as in here).

```{r, fig.width=6, fig.height=8}

target = read_tsv("./data/GO_term_glycolytic process.txt")
target = target$Symbol

# select only the targets to show on the heatmap

m = mats[rownames(mats)%in%target,]

targets = res_tbl %>% filter(gene%in%target)

# aheatmap(m, color = rev(brewer.pal(9,"RdBu")), annColors = "Set2", scale = "none")

newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="Glycolysis heatmap", 
         cluster_cols=T,fontsize_row=8, border_color=NA,
         labels_row = as.expression(newnames))

target

```

# Trpp Mrpl28 KO vs scramble
## Volcano plot 

```{r, fig.height = 15, fig.width = 15}

trpp_mrpl  %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey80',
                  fontface = "italic")+
  labs(title = "Mrpl28 KO cells vs scramble cells in Trp+")+
  theme_bw()+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Mrpl_vs_Scr/Trpp_Mrpl_vs_Scr_volcano.svg", dpi = 300,
       height = 10, width = 15, device = "svg")

```

Getting some stats yay

```{r, echo=TRUE}

table(res_tbl$padj<=0.05) # significant genes

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange<=-1) # signif and downreg

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange>=1) # signif and upreg

```

## GSEA analysis logFc

Nous n'inclurons pas l'analyse GSEA pour les Mrpl28 KO Trpp parce que les p-valeurs ne sont pas intéressantes.


## ORA
### Summary

```{r, fig.height = 5, fig.width = 7}

up = trpp_mrpl1 %>% filter(abs(log2FoldChange) >= 1)
up = up$gene

ora_results <- enricher(
  gene = up, 
  pvalueCutoff = 0.05, 
  pAdjustMethod = "BH", 
  TERM2GENE = dplyr::select(
    mm_GO_sets,
    gs_name,
    gene_symbol
  )
)

targetora = toupper(c("gobp_response_to_oxygen_levels",
                      "gobp_cellular_response_to_oxygen_levels",
                      "gobp_pyruvate_metabolic_process",
                      "gobp_atp_metabolic_process",
                      "gobp_adp_metabolic_process",
                      "gobp_nucelotide_phosphorylation",
                      "gobp_carbohydrate_catabolic_process",
                      "gobp_generation_of_precursor_metabolites_and_energy"))

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  dotplot(showCategory = 30, font.size = 8)+ 
  ggtitle("Over-representation analysis")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Mrpl_vs_Scr/Trpp_Mrpl_vs_Scr_ora_dotplot.svg", dpi = 300,
       height = 4, width = 6, device = "svg")

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  barplot(showCategory = 30, font.size = 8)+ 
  ggtitle("Over-representation analysis")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Mrpl_vs_Scr/Trpp_Mrpl_vs_Scr_ora_barplot.svg", dpi = 300,
       height = 4, width = 6, device = "svg")
 
ora_results %>%
  filter(ora_results@result$ID %in% targetora) %>% 
  as_tibble() %>%
  select(ID, GeneRatio,	p.adjust,	qvalue) %>%
  knitr::kable()

write.xlsx(ora_results, "./data_output/Trpp_Mrpl_vs_Scr/Trpp_Mrpl_vs_Scr_ora.xlsx")

```

## Heatmap : gene expression per condition

```{r}

dds = readRDS("./data_output/Trpp_Mrpl_vs_Scr/dds.rds")
diff = read.csv("./data_output/Trpp_Mrpl_vs_Scr/res_tbl_signif.csv")

reslog = rlog(dds, blind = FALSE)
mat = assay(reslog)

coldata = as.data.frame(colData(reslog))
colnames(mat) = rownames(coldata)

# Do the scaling to get z-scores

basemean = rowMeans(mat)
mats = t(apply(mat, 1, scale))
colnames(mats) = colnames(mat)

mats = mats[diff$ensembl,]
rownames(mats) = diff$gene

head(mats)
```


```{r, fig.width=6, fig.height=8}

precur = ora_results %>%
  as_tibble() %>%
  filter(ID == "GOBP_REGULATION_OF_GENERATION_OF_PRECURSOR_METABOLITES_AND_ENERGY") %>%
  select(geneID)

temp = unlist(c(strsplit(precur$geneID[[1]],"/")))

temp

reg_atp = ora_results %>%
  as_tibble() %>%
  filter(ID == "GOBP_REGULATION_OF_ATP_METABOLIC_PROCESS") %>%
  select(geneID)

temp = unlist(c(strsplit(reg_atp$geneID[[1]],"/")))

temp

# Select only the targets to show on the heatmap

m = mats[rownames(mats)%in%c("Me1", "Nos2", "Cox6a2", "Bnip3", "Ak4", "mt-Nd2", "Pgm1",
                             "P4ha2", "Gys1", "Gipr", "Gck", "Nupr1", "Pgam1", "Aldoc", 
                             "Pygl", "Slc4a4", "Pfkl", "Gbe1", "Bnip3","Mrpl28",
                             "Fam162a","Cxcr4","Gapdh"),]

# Set the rows of the heatmap in italics

newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="ATP and precursors metabolism heatmap", 
         cluster_cols=F,fontsize_row=8, border_color=NA,
         labels_row = as.expression(newnames))

targets = trpp_mrpl %>% filter(gene%in%c("Me1", "Nos2", "Cox6a2", "Bnip3", "Ak4", "mt-Nd2", "Pgm1",
                             "P4ha2", "Gys1", "Gipr", "Gck", "Nupr1", "Pgam1", "Aldoc", 
                             "Pygl", "Slc4a4", "Pfkl", "Gbe1", "Bnip3","Mrpl28",
                             "Fam162a","Cxcr4","Gapdh"))

```

All genes implicated in ATP metabolism, precursor generation and response to oxygen are the same ones.

```{r, fig.height=7, fig.width=8}

trpp_mrpl %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) > 1, 
             label = ifelse(padj<0.05&log2FoldChange>=1|
                              padj<0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "lightcoral")) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  theme_bw()+
  geom_point(data=targets, 
             aes(x=log2FoldChange,y=-log10(padj)), 
             color='blue',
             size=0.5)+
  geom_text_repel(data = targets, size = 3, segment.color = "blue",
                  max.overlaps = 30, min.segment.length = 0, color = "blue",
                  box.padding = 0.5)+
  theme(legend.title= element_blank())
```

## Heatmap with complete list of glycolysis genes

```{r, fig.width=6, fig.height=6}

target = read_tsv("./data/GO_term_glycolytic process.txt")
target = target$Symbol

# select only the targets to show on the heatmap

m = mats[rownames(mats)%in%target,]

targets = trpp_mrpl %>% filter(gene%in%target)

# aheatmap(m, color = rev(brewer.pal(9,"RdBu")), annColors = "Set2", scale = "none")

newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="Glycolysis heatmap", 
         cluster_cols=T,fontsize_row=8, border_color=NA,
         labels_row = as.expression(newnames))

target

```

# Trpp Ndufa2 KO vs scramble
## Volcano plot 

```{r, fig.height = 15, fig.width = 15}

trpp_nduf  %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) >= 1, 
             label = ifelse(padj<=0.05&log2FoldChange>=1|
                              padj<=0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "firebrick3")) +
  labs(col="Significantly expressed")+
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  geom_text_repel(max.overlaps = 15,
                  box.padding = 0.25,
                  segment.color = 'grey80',
                  fontface = "italic")+
  labs(title = "Ndufa2 KO cells vs scramble cells in Trp+")+
  theme_bw()+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Nduf_vs_Scr/Trpp_Mrpl_vs_Scr_volcano.svg", dpi = 300,
       height = 10, width = 15, device = "svg")

```

Getting some stats 

```{r, echo=TRUE}

table(res_tbl$padj<=0.05) # significant genes

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange<=-1) # signif and downreg

table(res_tbl$padj<=0.05&res_tbl$log2FoldChange>=1) # signif and upreg

```

## GSEA analysis logFc

Nous n'inclurons pas l'analyse GSEA pour les Ndufa2 KO Trpp parce que les p-valeurs ne sont pas intéressantes.

## ORA
### Summary

```{r, fig.height = 5, fig.width = 7}

up = trpp_nduf1 %>% filter(abs(log2FoldChange) >= 1)
up = up$gene

ora_results <- enricher(
  gene = up, 
  pvalueCutoff = 0.05, 
  pAdjustMethod = "BH", 
  TERM2GENE = dplyr::select(
    mm_all,
    gs_name,
    gene_symbol
  )
)

targetora = toupper(c("gobp_response_to_oxygen_levels",
                      "gobp_cellular_response_to_oxygen_levels",
                      "gobp_pyruvate_metabolic_process",
                      "gobp_atp_metabolic_process",
                      "gobp_adp_metabolic_process",
                      "gobp_nucelotide_phosphorylation",
                      "gobp_carbohydrate_catabolic_process",
                      "gobp_generation_of_precursor_metabolites_and_energy",
                      "gobp_nad_metabolic_process",
                      "gobp_nadh_metabolic_process",
                      "gobp_nadh_regeneration",
                      "gobp_glycolytic_process_through_fructose_6_phosphate",
                      "gobp_glucose_catabolic_process",
                      "reactome_metabolism_of_carbohydrates",
                      "reactome_glycolysis",
                      "reactome_glucose_metabolism",
                      "hallmark_hypoxia",
                      "hallmark_glycolysis",
                      "almmark_mtorc1_signaling",
                      "wp_glycolysis_and_gluconeogenesis",
                      "wp_aerobic_glycolysis"))

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  dotplot(showCategory = 30, font.size = 8)+ 
  ggtitle("Over-representation analysis")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Nduf_vs_Scr/Trpp_Nduf_vs_Scr_ora_dotplot.svg", dpi = 300,
       height = 7, width = 8, device = "svg")

ora_results %>% 
  filter(ora_results@result$ID %in% targetora) %>% 
  barplot(showCategory = 30, font.size = 8)+ 
  ggtitle("Over-representation analysis")+
   theme(axis.line = element_line(color='black'),
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

ggsave(plot = last_plot(), "./figures/Trpp_Nduf_vs_Scr/Trpp_Nduf_vs_Scr_ora_barplot.svg", dpi = 300,
       height = 4, width = 6, device = "svg")
 
ora_results %>%
  filter(ora_results@result$ID %in% targetora) %>% 
  as_tibble() %>%
  select(ID, GeneRatio,	p.adjust,	qvalue) %>%
  knitr::kable()

write.xlsx(ora_results, "./figures/Trpp_Nduf_vs_Scr/Trpp_Nduf_vs_Scr_ora.xlsx")

```

## Heatmap : gene expression per condition

```{r}

dds = readRDS("./data_output/Trpp_Nduf_vs_Scr/dds.rds")
diff = read.csv("./data_output/Trpp_Nduf_vs_Scr/res_tbl_signif.csv")

reslog = rlog(dds, blind = FALSE)
mat = assay(reslog)

coldata = as.data.frame(colData(reslog))
colnames(mat) = rownames(coldata)

# Do the scaling to get z-scores

basemean = rowMeans(mat)
mats = t(apply(mat, 1, scale))
colnames(mats) = colnames(mat)

mats = mats[diff$ensembl,]
rownames(mats) = diff$gene

head(mats)
```


```{r, fig.width=6, fig.height=8}

precur = ora_results %>%
  as_tibble() %>%
  filter(ID == "GOBP_GENERATION_OF_PRECURSOR_METABOLITES_AND_ENERGY") %>%
  select(geneID)

temp = unlist(c(strsplit(precur$geneID[[1]],"/")))

wp_glyc = ora_results %>%
  as_tibble() %>%
  filter(ID == "WP_GLYCOLYSIS_AND_GLUCONEOGENESIS") %>%
  select(geneID)

temp = unlist(c(strsplit(reg_atp$geneID[[1]],"/")))

# select only the targets to show on the heatmap

m = mats[rownames(mats)%in%c("Me1", "Nos2", "Cox6a2", "Bnip3", "Ak4", "mt-Nd2", "Pgm1",
                             "P4ha2", "Gys1", "Gipr", "Gck", "Nupr1", "Pgam1", "Aldoc", 
                             "Pygl", "Slc4a4", "Pfkl", "Gbe1", "Bnip3","Mrpl28",
                             "Fam162a","Cxcr4","Gapdh"),]

# Change the font to italics
newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="ATP and precursors metabolism heatmap", 
         cluster_cols=T,fontsize_row=10, border_color=NA, 
         legend = T,
         labels_row = as.expression(newnames))


targets = trpp_nduf %>% filter(gene%in%c("Me1", "Nos2", "Cox6a2", "Bnip3", "Ak4", "mt-Nd2", "Pgm1",
                             "P4ha2", "Gys1", "Gipr", "Gck", "Nupr1", "Pgam1", "Aldoc", 
                             "Pygl", "Slc4a4", "Pfkl", "Gbe1", "Bnip3","Mrpl28",
                             "Fam162a","Cxcr4","Gapdh"))

```



```{r, fig.height=7, fig.width=8}

trpp_nduf %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj),
             color = padj < 0.05 & abs(log2FoldChange) > 1, 
             label = ifelse(padj<0.05&log2FoldChange>=1|
                              padj<0.05&log2FoldChange<=-1,as.character(gene),''))) +
  scale_colour_manual(values = c("gray", "lightcoral")) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)+
  theme_bw()+
  geom_point(data=targets, 
             aes(x=log2FoldChange,y=-log10(padj)), 
             color='blue',
             size=0.5)+
  geom_text_repel(data = targets, size = 3, segment.color = "blue",
                  max.overlaps = 30, min.segment.length = 0, color = "blue",
                  box.padding = 0.5,
                  fontface = "italic")+
  theme(legend.title= element_blank())

```

## Heatmap with complete list of glycolysis genes

```{r, fig.width=6, fig.height=8}

target = read_tsv("./data/GO_term_glycolytic process.txt")
target = target$Symbol

# select only the targets to show on the heatmap

m = mats[rownames(mats)%in%target,]

targets = trpp_nduf %>% filter(gene%in%target)

# Change the fonts to italics

newnames <- lapply(
  rownames(m),
  function(x) bquote(italic(.(x))))

pheatmap(m, main="Glycolysis heatmap", 
         cluster_cols=T,fontsize_row=8, border_color=NA,
         labels_row = as.expression(newnames))

target

```

# Summary heatmap for oxphos and glycolysis
## OXPHOS
### Trp+ vs Trp-

```{r, fig.width=5, fig.height=5}

# concatenating significant genes from all conditions vs scramble cells

nonuc_trpn_trpp1$experiment = "Nonuc_Trpn_vs_Trpp"
mrpl_trpn_trpp1$experiment = "Mrpl28_Trpn_vs_Trpp"
nduf_trpn_trpp1$experiment = "Ndufa2_Trpn_vs_Trpp"

trpp_mrpl1$experiment = "Mrpl28_Trpp"
trpp_nduf1$experiment = "Ndufa2_Trpp"
trpp_nonuc1$experiment = "Nonuc_Trpp"

trpn_nonuc1$experiment = "Nonuc_Trpn"
trpn_nduf1$experiment = "Ndufa2_Trpn"
trpn_mrpl1$experiment = "Mrpl28_Trpn"

fc_trpn_trpp = 
  list(nonuc_trpn_trpp1, mrpl_trpn_trpp1, nduf_trpn_trpp1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

# determine targets

target = read_tsv("./data/GO_term_oxphos.txt")
target = target$Symbol

fc = fc_trpn_trpp %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene

fc_matrix = as.matrix(fc_matrix)

```

NAs replaced by 0s

```{r, fig.width=5, fig.height=7}

pheatmap(fc_matrix, main="Trp- vs Trp+", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")

```

### Nonuc and KOs vs scramble in Trp+ or Trp-

```{r, fig.width=5, fig.height=5}

# concatenating significant genes from all conditions vs scramble cells

fc_trpp = 
  list(trpp_nonuc1, trpp_mrpl1, trpp_nduf1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

fc = fc_trpp %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene
fc_matrix = as.matrix(fc_matrix)

p11 = pheatmap(fc_matrix, main="Nonuc and KOs vs scramble in Trpn+", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")

```


```{r, fig.width=5, fig.height=5}

# concatenating significant genes from all conditions vs scramble cells

fc_trpn = 
  list(trpn_nonuc1, trpn_mrpl1, trpn_nduf1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

# determine targets

target = c("Mgarp", "Pfkm")
target = sample(fc_trpn$gene, 60)

fc = fc_trpn %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene
fc_matrix = as.matrix(fc_matrix)

p12 = pheatmap(fc_matrix, main="Nonuc and KOs vs scramble in Trpn-", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")
```

```{r, fig.width=12, fig.height=12}

# Heatmaps objects are weird objects so the way to plot them aligned is
# more complicated than usual

grob = grid::grid.grabExpr(print(p12)) 
grob1 = grid::grid.grabExpr(print(p11)) 

plot_grid(grob, grob1, ncol = 2, align = "v")

```

## Glycolysis
### Trp+ vs Trp-

```{r, fig.width=5, fig.height=5}

fc_trpn_trpp = 
  list(nonuc_trpn_trpp1, mrpl_trpn_trpp1, nduf_trpn_trpp1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

# determine targets

target = read_tsv("./data/GO_term_glycolytic process.txt")
target = target$Symbol

fc = fc_trpn_trpp %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene
fc_matrix = as.matrix(fc_matrix)

```

NAs replaced by 0s

```{r, fig.width=5, fig.height=7}

pheatmap(fc_matrix, main="Trp 4 µM vs Trp 70 µM for 72h", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")

```

### Nonuc and KOs vs scramble in Trp+ or Trp-

```{r, fig.width=5, fig.height=5}

# concatenating significant genes from all conditions vs scramble cells

fc_trpp = 
  list(trpp_nonuc1, trpp_mrpl1, trpp_nduf1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

fc = fc_trpp %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene
fc_matrix = as.matrix(fc_matrix)

p13 = pheatmap(fc_matrix, main="Nonuc and KOs vs scramble in Trp+", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")

```


```{r, fig.width=5, fig.height=5}

# concatenating significant genes from all conditions vs scramble cells

fc_trpn = 
  list(trpn_nonuc1, trpn_mrpl1, trpn_nduf1) %>% 
  purrr::reduce(full_join) %>% 
  dplyr::select(starts_with("log2F"), description, gene, experiment) %>% 
  pivot_wider(names_from = experiment,
              values_from = log2FoldChange)

fc = fc_trpn %>% 
 filter(gene %in% target)

fc_matrix = fc[,3:5] %>% 
 replace(is.na(.), 0) 

rownames(fc_matrix) = fc$gene
fc_matrix = as.matrix(fc_matrix)

p14 = pheatmap(fc_matrix, main="Nonuc and KOs vs scramble in Trp-", cluster_cols=F, 
         fontsize_row=8, border_color=NA, scale = "none")
```

```{r, fig.width=12, fig.height=8}

# Heatmaps objects are weird objects so the way to plot them aligned is
# more complicated than usual

grob = grid::grid.grabExpr(print(p13)) 
grob1 = grid::grid.grabExpr(print(p14)) 

plot_grid(grob, grob1, ncol = 2, align = "v")
```

 